pkg/controllers/csi/provisioner/cleanup/appmounts.go

```diff
@@ -0,0 +1,57 @@
+package cleanup
+
+func (c *Cleaner) removeDeprecatedMounts(fsState fsState) {
+	stillMountedCounter := 0
+
+	for _, depDir := range fsState.deprecatedDks {
+		runDir := c.path.AgentRunDir(depDir)
+
+		volumeDirs, err := c.fs.ReadDir(runDir)
+		if err != nil {
+			log.Info("couldn't list volume dirs", "path", runDir)
+
+			continue
+		}
+
+		for _, volumeDir := range volumeDirs {
+			mappedDir := c.path.OverlayMappedDir(depDir, volumeDir.Name())
+
+			isEmpty, _ := c.fs.IsEmpty(mappedDir)
+			if isEmpty {
+				volumeDirPath := c.path.AgentRunDirForVolume(depDir, volumeDir.Name())
+
+				err := c.fs.RemoveAll(volumeDirPath)
+				if err == nil {
+					log.Info("removed unused volume", "path", volumeDirPath)
+
+					continue
+				}
+			}
+
+			stillMountedCounter++
+		}
+
+		isEmpty, _ := c.fs.IsEmpty(runDir)
+		if !isEmpty {
+			continue
+		}
+
+		err = c.fs.RemoveAll(runDir)
+		if err != nil {
+			log.Info("removed empty deprecated run dir", "path", runDir)
+
+			continue
+		}
+
+		tenantDir := c.path.DynaKubeDir(depDir)
+
+		err = c.fs.RemoveAll(c.path.DynaKubeDir(tenantDir))
+		if err != nil {
```

Comment:
Comparing to condition in line 24 and message in line 25
Should it be err== or err!= if followed by SUCCESSFUL? message "removed empty ... dir"